### PR TITLE
optimize:unify the config style and keep defaults consistent

### LINKED ISSUE #2222: optimize the default values for the seata-spring-boot-starter configuration item
- [x] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
There may be two different types of default values that need to be kept consistent. We want to be able to reference the same default value constant to increase maintainability.

image
image

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅵ. Environment:

- JDK version :
- OS :
- Others:

### PR DESCRIPTION
**Pull request by xingfudeshi**
### Ⅰ. Describe what this PR did
This PR uniformly converts the configuration style in seata to camel style, and uses the same reference for the default value of client configuration.

### Ⅱ. Does this pull request fix one issue?

fixes #2222 

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
I've done integration testing.

### CODE REVIEW

File: seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/file/ServiceProperties.java [hunk 1]
```diff
@@ -78,4 +81,14 @@ public ServiceProperties setDisableGlobalTransaction(boolean disableGlobalTransa
...
         return this;
     }
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        if (0 == vgroupMapping.size()) {
+            vgroupMapping.put("my_test_tx_group", "default");
+        }
+        if (0 == grouplist.size()) {
+            grouplist.put("default", "127.0.0.1:8091");
```
Comment (zjinlei) [inline seata-spring-boot-starter/src/main/java/io/seata/spring/boot/autoconfigure/properties/file/ServiceProperties.java]: use constant instead 127.0.0.1:8091

File: saga/seata-saga-engine-store/src/main/java/io/seata/saga/engine/config/DbStateMachineConfig.java [hunk 2]
```diff
@@ -43,7 +42,7 @@
     private String tablePrefix = "seata_";
     private String dbType;
     private SagaTransactionalTemplate sagaTransactionalTemplate;
-    private boolean                   rmReportSuccessEnable = ConfigurationFactory.getInstance().getBoolean(ConfigurationKeys.CLIENT_REPORT_SUCCESS_ENABLE, true);
+    private boolean rmReportSuccessEnable = ConfigurationFactory.getInstance().getBoolean(ConfigurationKeys.CLIENT_REPORT_SUCCESS_ENABLE, true);
```
[the remark this answers is not in the document]
Comment (xingfudeshi) [reply saga/seata-saga-engine-store/src/main/java/io/seata/saga/engine/config/DbStateMachineConfig.java]: done.

[the remark this answers is not in the document]
Comment (xingfudeshi) [reply saga/seata-saga-engine-store/src/main/java/io/seata/saga/engine/config/DbStateMachineConfig.java]: done

File: spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java [hunk 3]
```diff
@@ -47,7 +47,8 @@
  *
  * @author slievrly
  */
-public class GlobalTransactionalInterceptor implements ConfigurationChangeListener,MethodInterceptor {
+public class GlobalTransactionalInterceptor implements ConfigurationChangeListener, MethodInterceptor {
+    public static final boolean DEFAULT_DISABLE_GLOBAL_TRANSACTION = false;
```
Comment (zjinlei) [inline spring/src/main/java/io/seata/spring/annotation/GlobalTransactionalInterceptor.java]: move to DefaultValues

File: core/src/main/java/io/seata/core/constants/ConfigurationKeys.java [hunk 4]
```diff
@@ -154,22 +162,22 @@
...
      * The constant STORE_DB_GLOBAL_DEFAULT_TABLE.
      */
-    public static final String STORE_DB_GLOBAL_DEFAULT_TABLE = "global_table";
+    public static final String STORE_DB_GLOBAL_DEFAULT_TABLE = "globalTable";
 
     /**
      * The constant STORE_DB_BRANCH_DEFAULT_TABLE.
      */
-    public static final String STORE_DB_BRANCH_DEFAULT_TABLE = "branch_table";
+    public static final String STORE_DB_BRANCH_DEFAULT_TABLE = "branchTable";
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/constants/ConfigurationKeys.java]: globalTable->global_table, branchTable->branch_table
and move them to DefaultValues.java

File: core/src/main/java/io/seata/core/constants/DefaultValues.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+ * @author xingfudeshi@gmail.com
+ */
+public class DefaultValues {
+    public static final int DEFAULT_CLIENT_LOCK_RETRY_INTERNAL = 10;
+    public static final int DEFAULT_CLIENT_LOCK_RETRY_TIMES = 30;
+    public static final boolean DEFAULT_CLIENT_LOCK_RETRY_POLICY_BRANCH_ROLLBACK_ON_CONFLICT = true;
+    public static final int DEFAULT_LOG_EXCEPTION_RATE = 100;
+    public static final int DEFAULT_CLIENT_ASYNC_COMMIT_BUFFER_LIMIT = 10000;
+    public static final int DEFAULT_CLIENT_REPORT_RETRY_COUNT = 5;
+    public static final boolean DEFAULT_CLIENT_REPORT_SUCCESS_ENABLE = true;
```
[the remark this answers is not in the document]
Comment (xingfudeshi) [reply core/src/main/java/io/seata/core/constants/DefaultValues.java]: I think it can be set to false as default.

**Comment by xingfudeshi** [comment]
> Did you forget to modify
> 
> StarterConstants#THREAD_FACTORY_PREFIX
I added a new constant for ThreadFactory in kebab style as spring boot demands.

https://docs.spring.io/spring-boot/docs/2.1.7.RELEASE/reference/html/boot-features-external-config.html  
image

File: script/client/spring/application.properties [hunk 6]
```diff
@@ -32,8 +32,8 @@ seata.client.undo.data-validation=true
 seata.client.undo.log-serialization=jackson
 seata.client.undo.log-table=undo_log
 seata.client.log.exceptionRate=100
-seata.service.vgroup-mapping=default
-seata.service.grouplist=127.0.0.1:8091
+seata.service.vgroup-mapping.my_test_tx_group=default
+seata.service.grouplist.default=127.0.0.1:8091
 seata.service.enable-degrade=false
 seata.service.disable-global-transaction=false
 seata.transport.shutdown.wait=3
```
Comment (zjinlei) [inline script/client/spring/application.properties]: application.properties and application.yml underlineToCamel

**Comment by slievrly** [comment]
> seata.registry.etcd3.serverAddr is null

it's my fault.

File: script/client/spring/application.properties (hunk 6 above)
Comment (xingfudeshi) [reply script/client/spring/application.properties]: Not required.both camel and kebab style can be supported by springboot.

**Comment by slievrly** [comment]
remove `seata.config.file.name`  and `seata.registry.file.name`

File: core/src/main/java/io/seata/core/constants/ConfigurationKeys.java [hunk 7]
```diff
@@ -203,27 +201,27 @@
...
      * The constant LOCK_DB_TABLE.
      */
-    public static final String LOCK_DB_TABLE = STORE_DB_PREFIX + "lock-table";
+    public static final String LOCK_DB_TABLE = STORE_DB_PREFIX + "lockTable";
 
     /**
      * The constant LOCK_DB_DEFAULT_TABLE.
      */
-    public static final String LOCK_DB_DEFAULT_TABLE = "lock_table";
+    public static final String LOCK_DB_DEFAULT_TABLE = "lockTable";
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/constants/ConfigurationKeys.java]: lockTable-> lock_table
and move to DeafultValues

File: core/src/main/java/io/seata/core/constants/ConfigurationKeys.java [hunk 8]
```diff
@@ -88,51 +88,59 @@
...
     /**
      * The constant CLIENT_LOCK_RETRY_TIMES.
      */
-    public static final String CLIENT_LOCK_RETRY_TIMES = CLIENT_RM_PREFIX + "lock.retry.times";
+    public static final String CLIENT_LOCK_RETRY_TIMES = CLIENT_RM_LOCK_PREFIX + "retryTimes";
     /**
      * The constant CLIENT_LOCK_RETRY_INTERNAL.
      */
-    public static final String CLIENT_LOCK_RETRY_INTERNAL = CLIENT_RM_PREFIX + "lock.retry.internal";
+    public static final String CLIENT_LOCK_RETRY_INTERNAL = CLIENT_RM_LOCK_PREFIX + "retryInternal";
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/constants/ConfigurationKeys.java]: internal-> interval
Multiple places need to be modified

File: core/src/main/java/io/seata/core/constants/ConfigurationKeys.java (hunk 7 above)
Comment (slievrly) [reply core/src/main/java/io/seata/core/constants/ConfigurationKeys.java]: great found.

**Pull request merged**